sklearn/manifold/tests/test_spectral_embedding.py

```diff
@@ -42,7 +42,7 @@ def _check_with_col_sign_flipping(A, B, tol=0.0):
                          (((A[:, column_idx] +
                             B[:, column_idx]) ** 2).mean() <= tol ** 2))
         if not sign:
-            return False
```

Comment:
better to change this function to be `assert_equal_with_sign_flipping`.

The function is awkward because it keeps `sign` from iteration to iteration, but we know at the start of each iteration that `sign` is True.

So it would be sufficient to have something like
```py
def ...
    for A_col, B_col in zip(A.T, B.T):
        assert np.mean((A_col - B_col) ** 2) <= tol ** 2 or np.mean((A_col + B_col) ** 2) <= tol ** 2
```

Reply:
Nice. I'll make it `np.max` though, as we might miss outliers otherwise.